src/App.tsx

```diff
@@ -17,18 +52,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todoList={todoList}
+                onFilterChange={handleFilterChange}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todoList={filteredTodoList}
+                  onEyeClick={toggleModal}
+                  isModalOpen={isModalOpen}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalOpen && (
+        <TodoModal
+          toggleModal={toggleModal}
```

Comment:
you can just pass the closeModal prop
```suggestion
          closeModal={() => setSelectedTodo(null)}
```